packages/editor-ui/src/components/canvas/elements/nodes/render-types/parts/CanvasNodeDisabledStrikeThrough.vue

```diff
@@ -6,16 +6,23 @@ import { useCanvasNode } from '@/composables/useCanvasNode';
 const $style = useCssModule();
 
 const { inputs, outputs, connections } = useCanvasNode();
-const { mainInputConnections, mainOutputConnections } = useNodeConnections({
-	inputs,
-	outputs,
-	connections,
-});
-
-const isVisible = computed(
-	() => mainInputConnections.value.length === 1 && mainOutputConnections.value.length === 1,
+const { mainInputConnections, mainInputs, mainOutputConnections, mainOutputs } = useNodeConnections(
```

Comment:
i think you can accomplish the same with the information provided by https://vueflow.dev/typedocs/interfaces/Actions.html#findnode -> handleBounds

Reply:
That's definitely a possibility! Think there's an advantage to using that? I was thinking to keep parts without vueflow as a dependency as much as possible, so we can setup a storybook and testing easier.

Reply:
i would honestly move the logic visible / hidden to the parent component and make this component as dumb as possible.

in terms of dependencies , vueFlow is already a dependecy and since what gives us trouble is our own state i would try to remove that

Reply:
I've moved the strike-through condition into the node as suggested. Thanks!